R/config.R

```diff
@@ -0,0 +1,796 @@
+
+  tool_path <- tools_file[[tool_name]]
+
+  if (is.null(tool_path)) {
+    warning("Attribute does not exist.")
+  }
+
+  return(tool_path)
+}
+
+#' Get Parsed Config File
+#'
+#' Gets the parsed configuration file for any path-specified project
+#'
+#' @param config_path The path of the config file from the kaiaulu directory (e.g. "conf/kaiaulu.yml")
+#' @return the parsed config file whose path is specified by `config_path`
+#' @export
+get_conf <- function(config_path) {
+
+  conf <- yaml::read_yaml(config_path)
+
+  if (is.null(conf)) {
+    warning("Path does not exist.")
+  }
+
+  return(conf)
+}
+
+#' Get Project Git Repo Path
+#'
+#' Gets the local project git repository path of the project
+#'
+#' @param config_file The parsed config file
+#' @return the local git repository path specified in `config_file`
+#' @export
+get_git_repo_path <- function(config_file) {
+
+  git_repo_path <- config_file[["version_control"]][["log"]]
+
+  if (is.null(git_repo_path)) {
+    warning("Attribute does not exist in the configuration file.")
+  }
+
+  return(git_repo_path)
+}
+
+#' Get Project Git Branches
+#'
+#' Gets the list of git branches of the project
+#'
+#' @param config_file The parsed config file
+#' @return the list of git branches
+#' @export
+get_git_branches <- function(config_file) {
+
+  git_branch <- config_file[["version_control"]][["branch"]]
+
+  if (is.null(git_branch)) {
+    warning("Attribute does not exist in the configuration file.")
+  }
+
+  return(git_branch)
+}
+
+#' Get File Extensions
+#'
+#' Gets the file extensions for filtering
+#'
+#' @param config_file The parsed config file
+#' @return the file extensions
+#' @export
+get_file_extensions <- function(config_file) {
+
+  file_extensions <- config_file[["filter"]][["keep_filepaths_ending_with"]]
+
+  if (is.null(file_extensions)) {
+    warning("Attribute does not exist in the configuration file.")
+  }
+
+  return(file_extensions)
+}
+
+#' Get Substring Filepaths
+#'
+#' Gets the substring filepaths for filtering
+#'
+#' @param config_file The parsed config file
+#' @return the substring filepaths
+#' @export
+get_substring_filepath <- function(config_file) {
+
+  substring_filepath <- config_file[["filter"]][["remove_filepaths_containing"]]
+
+  if (is.null(substring_filepath)) {
+    warning("Attribute does not exist in the configuration file.")
+  }
+
+  return(substring_filepath)
+}
+
+#' Get Filter Commit Size
+#'
+#' Gets the commit size to filter out
+#'
+#' @param config_file The parsed config file
+#' @return the commit size to filter out
+#' @export
+get_filter_commit_size <- function(config_file) {
+
+  filter_commit_size <- config_file[["filter"]][["remove_filepaths_on_commit_size_greather_than"]]
+
+  if (is.null(filter_commit_size)) {
+    warning("Attribute does not exist in the configuration file.")
+  }
+
+  return(filter_commit_size)
+}
+
+#' Get Uctags Line Types
+#'
+#' Gets the Uctags keep lines type
```

Comment:
Also here. I do not know what this means, so I assume there are others who would also not know.